### PR TITLE
Adds hlint rules for `asumMap`

### LINKED ISSUE #86: Add HLint rules for `asumMap` function
The `asumMap` function was recently introduced in #85:

https://github.com/kowainik/relude/blob/f69d531eafb7f397fa3103e9b811bd8f965a9108/src/Relude/Foldable/Fold.hs#L84-L85

It would be also great to add HLint rules for this function. I think it should be possible to come up with at least 3 different rules.

### PR DESCRIPTION
**Pull request by cronokirby**
Resolves #86 

### HLint

- [ ] I've changed the exposed interface (add new reexports, remove reexports, rename reexported things, etc.).
  - [x] I've updated [`hlint.dhall`](https://github.com/kowainik/relude/blob/master/hlint/hlint.dhall)
        accordingly to my changes (add new rules for the new imports, remove old ones, when they are outdated, etc.).
  - [x] I've generated the new `.hlint.yaml` file (see 
        [this instructions](https://github.com/kowainik/relude#generating-hlintyaml)).

### General

- [ ] I've updated the [CHANGELOG](https://github.com/kowainik/relude/blob/master/CHANGELOG.md) 
      with the short description of my latest changes.
- [ ] All new and existing tests pass.
- [ ] I keep the code style used in the files I've changed (see 
      [style-guide](https://github.com/kowainik/org/blob/master/style-guide.md#haskell-style-guide) 
      for more details).
- [ ] I've used the [`stylish-haskell` file](https://github.com/kowainik/relude/blob/master/.stylish-haskell.yaml).
- [ ] My change requires the documentation updates.
  - [ ] I've updated the documentation accordingly.
- [ ] I've added the `[ci skip]` text to the docs-only related commit's name.

### CODE REVIEW

File: hlint/hlint.dhall [hunk 1]
```diff
@@ -76,6 +76,10 @@ in [ rule.Arguments { arguments =
    , hintNote "fmap or (mapM f s)" "anyM f s" "Applying this hint would mean that some actions that were being executed previously would no longer be executed."
    , hintNote "or <$> mapM f s" "anyM f s" "Applying this hint would mean that some actions that were being executed previously would no longer be executed."
 
+   , warnSimple "getAlt . foldMap (Alt . f)" "asumMap"
+   , hintNote "foldr (\\x acc -> f x <|> acc) empty" "asumMap f" "Use 'asumMap'"
+   , hintNote "asum (map f xs)" "asumMap f xs" "Use 'asumMap'"
```
Comment (ChShersh) [inline hlint/hlint.dhall]: Cool separation on `warn` and `hint` :+1:

File: hlint/hlint.dhall [hunk 2]
```diff
@@ -76,6 +76,10 @@ in [ rule.Arguments { arguments =
    , hintNote "fmap or (mapM f s)" "anyM f s" "Applying this hint would mean that some actions that were being executed previously would no longer be executed."
    , hintNote "or <$> mapM f s" "anyM f s" "Applying this hint would mean that some actions that were being executed previously would no longer be executed."
 
+   , warnSimple "getAlt . foldMap (Alt . f)" "asumMap"
```
Comment (ChShersh) [inline hlint/hlint.dhall]: I think here it should be `"getAlt (foldMap (Alt . f) xs" "asumMap xs"`

As I recall correctly, `hlint` doesn't handle function composition properly... So by writing `f . g` it won't catch `f (g x)` cases

Comment (cronokirby) [reply hlint/hlint.dhall]: Didn't know about that, thanks

Comment (vrom911) [reply hlint/hlint.dhall]: I guess there should be both rules.. The same issues applied vice versa

File: hlint/hlint.dhall [hunk 3]
```diff
@@ -76,6 +76,11 @@ in [ rule.Arguments { arguments =
    , hintNote "fmap or (mapM f s)" "anyM f s" "Applying this hint would mean that some actions that were being executed previously would no longer be executed."
    , hintNote "or <$> mapM f s" "anyM f s" "Applying this hint would mean that some actions that were being executed previously would no longer be executed."
 
+   , warnSimple "getAlt (foldMap (Alt . f)) xs" "asumMap xs"
```
Comment (ChShersh) [inline hlint/hlint.dhall]: Oops, my comment was badly formatted and this probably lead to possible error. The parenthesis here are wrong, it should be `"getAlt (foldMap (Alt . f) xs)"`.

**Pull request merged**